src/realm/sync/network/default_socket.hpp

```diff
@@ -84,27 +80,106 @@ class EZObserver {
+        {
+            // Temporary workaround until the event loop is updated to use Status
+            auto&& new_handler = [handler = std::move(handler)](std::error_code ec) {
+                Status status = !ec ? Status::OK()
+                                : ec == util::error::operation_aborted
+                                    ? Status{ErrorCodes::OperationAborted, "Timer canceled"}
+                                    : Status{ErrorCodes::RuntimeError, "Timer failed"};
+                handler(status);
+            };
+            m_timer.async_wait(delay, std::move(new_handler));
+        }
+
+        /// Cancels the timer and destroys the timer instance.
+        ~Timer() = default;
+
+        /// Cancel the timer immediately
+        void cancel() override
+        {
+            m_timer.cancel();
+        }
+
+    private:
+        network::DeadlineTimer m_timer;
+    };
+
+    DefaultSocketProvider(const std::shared_ptr<util::Logger>& logger, const std::string user_agent)
+        : m_logger_ptr{logger}
+        , m_service{std::make_shared<network::Service>()}
+        , m_random{}
+        , m_user_agent{user_agent}
     {
-        REALM_ASSERT(m_config.logger); // Make sure the logger is valid
+        REALM_ASSERT(m_logger_ptr);                     // Make sure the logger is valid
+        REALM_ASSERT(m_service);                        // Make sure the service is valid
+        util::seed_prng_nondeterministically(m_random); // Throws
+        start_keep_running_timer();
     }
 
-    EZSocketFactory(EZSocketFactory&&) = delete;
+    // Don't allow move or copy constructor
+    DefaultSocketProvider(DefaultSocketProvider&&) = delete;
+
+    // Temporary workaround until event loop is completely moved here
+    network::Service& get_service()
+    {
+        return *m_service;
+    }
 
-    std::unique_ptr<EZSocket> connect(EZObserver* observer, EZEndpoint&& endpoint);
+    // Temporary workaround until Client::Connection is updated to use WebSocketObserver
+    std::unique_ptr<DefaultWebSocket> connect_legacy(EZObserver* observer, EZEndpoint&& endpoint);
+
+    std::unique_ptr<WebSocketInterface> connect(WebSocketObserver*, WebSocketEndpoint&&) override
+    {
+        return nullptr;
+    }
+
+    void post(FunctionHandler&& handler) override
+    {
+        REALM_ASSERT(m_service);
+        // Don't post empty handlers onto the event loop
+        if (handler) {
+            // Temporary workaround until the event loop is updated to use Status
+            auto&& new_handler = [handler = std::move(handler)]() {
+                handler(Status::OK());
+            };
+            m_service->post(std::move(new_handler));
+        }
+    }
+
+    SyncTimer create_timer(std::chrono::milliseconds delay, FunctionHandler&& handler) override
+    {
+        return std::make_unique<DefaultSocketProvider::Timer>(*m_service, delay, std::move(handler));
+    }
 
 private:
-    EZConfig m_config;
+    void start_keep_running_timer()
```

Comment:
Can we add a comment that we should remove/fix this so we don't have a timer scheduled for 1000 hours? Def fine to keep for now while we're moving things around though.